050-bug17561-reserveslip-needs-itemnumber.patch

This patch fixes the hold slip when multiple holds are made by same patron on same biblio.
It adds the itemnumber as param to request so proper hold slip for right reserve can be generated.

diff --git a/C4/Reserves.pm b/C4/Reserves.pm
index ae1c7b7..68c608d 100644
--- a/C4/Reserves.pm
+++ b/C4/Reserves.pm
@@ -2339,7 +2339,7 @@ sub GetReserveId {
 
 =head2 ReserveSlip
 
-  ReserveSlip($branchcode, $borrowernumber, $biblionumber)
+  ReserveSlip($branchcode, $borrowernumber, $biblionumber [, $itemnumber, $barcode ])
 
 Returns letter hash ( see C4::Letters::GetPreparedLetter ) or undef
 
@@ -2356,13 +2356,15 @@ available within the slip:
 =cut
 
 sub ReserveSlip {
-    my ($branch, $borrowernumber, $biblionumber) = @_;
+    my ($branch, $borrowernumber, $biblionumber, $itemnumber, $barcode) = @_;
 
 #   return unless ( C4::Context->boolean_preference('printreserveslips') );
+    my $item = C4::Items::GetItem($itemnumber, $barcode);
 
     my $reserve_id = GetReserveId({
         biblionumber => $biblionumber,
-        borrowernumber => $borrowernumber
+        borrowernumber => $borrowernumber,
+        itemnumber => $item->{itemnumber}
     }) or return;
     my $reserve = GetReserveInfo($reserve_id) or return;
 
diff --git a/circ/returns.pl b/circ/returns.pl
index d8efcc3..c9911dc 100644
--- a/circ/returns.pl
+++ b/circ/returns.pl
@@ -78,6 +78,7 @@ if ( $query->param('print_slip') ) {
         print_slip     => 1,
         borrowernumber => scalar $query->param('borrowernumber'),
         biblionumber   => scalar $query->param('biblionumber'),
+        itemnumber     => scalar $query->param('itemnumber'),
     );
 }

diff --git a/circ/hold-transfer-slip.pl b/circ/hold-transfer-slip.pl
index bd09e21..70feb06 100755
--- a/circ/hold-transfer-slip.pl
+++ b/circ/hold-transfer-slip.pl
@@ -38,6 +38,8 @@ my $session = get_session($sessionID);
 
 my $biblionumber = $input->param('biblionumber');
 my $borrowernumber = $input->param('borrowernumber');
+my $itemnumber = $input->param('itemnumber');
+my $barcode = $input->param('barcode');
 
 my ( $template, $loggedinuser, $cookie ) = get_template_and_user(
     {   
@@ -52,7 +54,7 @@ my ( $template, $loggedinuser, $cookie ) = get_template_and_user(
 
 my $userenv = C4::Context->userenv;
 my ($slip, $is_html);
-if ( my $letter = ReserveSlip ($session->param('branch') || $userenv->{branch}, $borrowernumber, $biblionumber) ) {
+if ( my $letter = ReserveSlip ($session->param('branch') || $userenv->{branch}, $borrowernumber, $biblionumber, $itemnumber, $barcode) ) {
     $slip = $letter->{content};
     $is_html = $letter->{is_html};
 }
diff --git a/koha-tmpl/intranet-tmpl/prog/en/modules/circ/circulation.tt b/koha-tmpl/intranet-tmpl/prog/en/modules/circ/circulation.tt
index a937bf1..1303ed0 100644
--- a/koha-tmpl/intranet-tmpl/prog/en/modules/circ/circulation.tt
+++ b/koha-tmpl/intranet-tmpl/prog/en/modules/circ/circulation.tt
@@ -372,7 +372,7 @@ $(document).ready(function() {
     <input type="hidden" name="borrowernumber" value="[% borrowernumber %]" />
     <input type="hidden" name="duedatespec" value="[% duedatespec %]" />
     <input type="hidden" name="stickyduedate" value="[% stickyduedate %]" />
-    <button class="print" type="submit" onclick="Dopop('hold-transfer-slip.pl?borrowernumber=[% reserveborrowernumber %]&amp;biblionumber=[% itembiblionumber %]&amp;op=slip');this.form.submit();"><i class="fa fa-print"></i> Don't check out and print slip (P)</button>
+    <button class="print" type="submit" onclick="Dopop('hold-transfer-slip.pl?borrowernumber=[% reserveborrowernumber %]&amp;biblionumber=[% itembiblionumber %]&amp;itemnumber=[% itemnumber %]&amp;op=slip');this.form.submit();"><i class="fa fa-print"></i> Don't check out and print slip (P)</button>
 </form>
 [% END %]
 
@@ -382,7 +382,7 @@ $(document).ready(function() {
     <input type="hidden" name="borrowernumber" value="[% borrowernumber %]" />
     <input type="hidden" name="duedatespec" value="[% duedatespec %]" />
     <input type="hidden" name="stickyduedate" value="[% stickyduedate %]" />
-    <button class="print" type="submit" onclick="Dopop('hold-transfer-slip.pl?borrowernumber=[% reserveborrowernumber %]&amp;biblionumber=[% itembiblionumber %]&amp;op=slip');this.form.submit();"><i class="fa fa-print"></i> Don't check out and print slip (P)</button>
+    <button class="print" type="submit" onclick="Dopop('hold-transfer-slip.pl?borrowernumber=[% reserveborrowernumber %]&amp;biblionumber=[% itembiblionumber %]&amp;itemnumber=[% itemnumber %]&amp;op=slip');this.form.submit();"><i class="fa fa-print"></i> Don't check out and print slip (P)</button>
 </form>
 [% END %]
 
diff --git a/koha-tmpl/intranet-tmpl/prog/en/modules/circ/returns.tt b/koha-tmpl/intranet-tmpl/prog/en/modules/circ/returns.tt
index c0b9c90..6b526e4 100644
--- a/koha-tmpl/intranet-tmpl/prog/en/modules/circ/returns.tt
+++ b/koha-tmpl/intranet-tmpl/prog/en/modules/circ/returns.tt
@@ -45,7 +45,7 @@ $(document).ready(function () {
     });
 
     [% IF print_slip %]
-        Dopop('hold-transfer-slip.pl?borrowernumber=[% borrowernumber %]&amp;biblionumber=[% biblionumber %]');
+        Dopop('hold-transfer-slip.pl?borrowernumber=[% borrowernumber %]&amp;biblionumber=[% biblionumber %]&amp;itemnumber=[% itemnumber %]');
     [% END %]
 
     var columns_settings = [% ColumnsSettings.GetColumns( 'circ', 'returns', 'checkedintable', 'json' ) %]
